service/api/pokemon_test.go

```diff
@@ -0,0 +1,38 @@
+package api
+
+import (
+	"bootcamp/domain/model"
+	"testing"
+)
+
+var fetchPokeApiMock func() (pokemon model.FetchedPokemon, err error)
+
+type PokeApiServiceMock struct{}
+
+func (s PokeApiServiceMock) fetchPokeApi(name string) (model.FetchedPokemon, error) {
+	return fetchPokeApiMock()
+}
+
+func TestFetchPokemon(t *testing.T) {
+	s = PokeApiServiceMock{}
+	someInt := 1
+	fetchedPokemon := model.FetchedPokemon{
+		Id:     someInt,
+		Forms:  []map[string]string{{"foo": "bar"}},
+		Height: someInt,
+		Name:   "Pikachu",
+		Order:  someInt,
+		Weight: someInt}
+
+	fetchPokeApiMock = func() (pokemon model.FetchedPokemon, err error) {
+		return fetchedPokemon, nil
+	}
+
+	res, err := FetchPokemon("")
+
+	if err != nil {
```

Comment:
When testing, make sure  that you define per case what you're expecting for a particular input and compare it with what you actually got.

I would suggest using the assert package for this https://pkg.go.dev/github.com/stretchr/testify/assert

Reply:
🆗